tools/ReadConsoleInputStream/ReadConsoleInputStream.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Diagnostics;
+using System.IO;
+
+using Vanara.PInvoke;
+
+namespace Nivot.Terminal
+{
+    /// <summary>
+    /// Provides a Stream-oriented view over the console's input buffer key events
+    /// while also collecting out of band events like buffer resize, menu etc in
+    /// a caller-provided BlockingCollection.
+    /// </summary>
+    /// <remarks>The buffer contains unicode chars, not 8 bit CP encoded chars as we rely on ReadConsoleInputW.</remarks>
+    public sealed class ReadConsoleInputStream : Stream
+    {
+        private const int BufferSize = 256;
```

Comment:
> BufferSize = 256; [](start = 25, length = 18)

Does this have to somehow match the other magic 256 used in the other file, or is it just coincidence?

Reply:
Just a coincidence. But I'll constantinople that one too.